demo/api/src/app.module.ts

```diff
@@ -79,6 +81,9 @@ export class AppModule {
                         return user.domains.includes(requestScope.domain);
                     },
                 }),
+                UserPermissionsModule.forRoot({
+                    config: UserPermissionsConfigService,
```

Comment:
With the current approach, the UserPermissionsModule in the library provides the service defined in the application. Shouldn't the application provide the service itself?

Reply:
I don't understand this. Do you think that making the config to a provider in the module (https://github.com/vivid-planet/comet/pull/1261/files#diff-87c5a6413e447c9c46e24a82309ca67e8200fd20332af148f99c70344145a287R26) is not necessary, instead the config should be directly injected into the UserPermissionsService?

Reply:
Usually, a dynamic module provides multiple methods to create a new instance, for instance `UserPermissionsModule.forRoot` and `UserPermissionsModule.forRootAsync`.  All methods expect the same outcome, for instance in this case, that would be an object (or class) containing the methods defined in `UserPermissionConfigInterface`. The module in the library doesn't care how this object was created. In the app it could be either a hard coded object (similar to the static authed user), or it could be a "real" Nest service provided by another module, that is then dynamically injected using `forRootAsync`. The library module would then provide the object, and only use the object it provides. In the case of a real service the service would be provided two times with different names

I think this would be the "more Nest-like" way. Do you understand what I'm trying to say? It's not easy for me to explain it 😅

Reply:
I think the "only" confusing thing here is that "config" is acutally an property of the module config (config inside the config)

That naming should be improved.